CommonMark/Parser/InlineMethods.cs

```diff
@@ -14,11 +14,12 @@ internal static class InlineMethods
         /// Initializes the array of delegates for inline parsing.
         /// </summary>
         /// <returns></returns>
-        internal static Func<Subject, Inline>[] InitializeParsers(CommonMarkSettings settings)
+        internal static Func<Subject, CommonMarkSettings, Inline>[] InitializeParsers(CommonMarkSettings settings)
```

Comment:
see the comment in CommonMarkSettings about not passing the settings to the delegates
it seems that you might need to do something like `p['~'] = (subject, inline) => HandleTilde(subject, inline, tildeOptions)`